### ISSUE TITLE
Hide turn restriction UI from `highway=crossing` for footway

### ISSUE THREAD

**Issue by tordans** [openstreetmap/iD#10222]
### URL

https://ideditor.netlify.app/#background=Bing&disable_features=boundaries&id=n8677399770&locale=en&map=20.00/52.28422/13.53982

### How to reproduce the issue?

1. open https://ideditor.netlify.app/#background=Bing&disable_features=boundaries&id=n8677399770&locale=en&map=20.00/52.28422/13.53982
2. see the turn restriction UI which does not make any sense for this node 
   <img width="946" alt="image" src="https://github.com/openstreetmap/iD/assets/111561/fb09b9bd-8e97-45f9-9527-ae74bf6fbc0e">

This actually allows to create "no u turn" restrictions on the crossing node:

<img width="749" alt="image" src="https://github.com/openstreetmap/iD/assets/111561/fa3b5f7e-5759-4a89-ab83-5ee98d265334">

### Which deployed environments do you see the issue in?

Development version at ideditor.netlify.app

### Which browsers are you seeing this problem on?

Chrome

**Comment by RudyTheDev** [comment]
I don't think this is related to crossings specifically. You can get the turn restriction UI for any place where `highway`s meet even when those `highway`s are not supposed to appear in the UI.

**Comment by Dimitar5555** [comment]
See #9983

**Comment by tordans** [comment]
> See #9983

Thanks @Dimitar5555 this could very well be a regression in https://github.com/openstreetmap/iD/commit/ea47581bae335ace9f117216c1882da8a26b4852
~Looking at the same node in Rapid (which does not have this change) the Turn-Restriction UI does not show up https://rapideditor.org/canary#map=20.00/52.28422/13.53982&background=Bing&datasets=fbRoads,msBuildings&disable_features=boundaries&id=n8677399770&locale=en~
- Update: I learned that Rapid does not have the turn restriction UI at all … so that test is not helpful :-)

@tyrasd I think ea47581bae335ace9f117216c1882da8a26b4852 should be reverted and the corresponding issue reopened. Having the UI show up too often is worse then having an edge case where it does not show at all, IMO.

**Comment by tyrasd** [comment]
> Having the UI show up too often is worse then having an edge case

It could be possible to hide the field in such cases, but still make it available to add manually via the _Add field_ box.

**Comment by matkoniecz** [comment]
note that some turn restrictions into/from footways may make sense and apply to cyclists
